File: RefMan/Client/src/models/Node.ts
export class Node {
    private _name: string;

    private _parent: Node | null = null;

    constructor(name: string) {
      this._name = name;
    }

    public get name(): string {
      return this._name;
    }

    public set name(value: string) {
      this._name = value;
    }

    public get parent(): Node | null {
      return this._parent;
    }

    public set parent(value: Node | null) {
      this._parent = value;
    }
}

File: RefMan/Models/FileSystem/Node.cs
﻿namespace RefMan.Models.FileSystem
{
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    using RefMan.Models.User;

    public class Node : EntryName
    {
        [Key]
        public long Id { get; set; }

        public bool IsRoot { get; set; }

        [ForeignKey(nameof(Parent))]
        public long? ParentId { get; set; }

        public Folder Parent { get; set; }

        [ForeignKey(nameof(Owner))]
        public long OwnerId { get; set; }

        public AppUser Owner { get; set; }
    }
}